activesupport/lib/active_support/cache.rb

```diff
@@ -817,6 +817,12 @@ def namespace_key(key, options = nil)
           end
         end
 
+        # Validates the expanded key.
+        # Raises an exception when the key is +nil+ or an empty string.
+        def validated_key(key)
+          key.presence || raise(ArgumentError, "key cannot be blank")
+        end
```

Comment:
I don't think this really deserve an extracted method. I'd suggest poutting this logic in `normalize_key`.

Reply:
Thanks @byroot I've moved it into `normalize_key`.